### PR TITLE
Antoine's solution

### PR DESCRIPTION
**Pull request by poudro**
Here's my approach to your challenge.

I use python's multiprocessing module to be able to control the number of processes running (use all cores of cpu). First of all, this requires to transform the file objects into file-like objects that are passed to the child processes as file descriptors cannot be passed this way.

Also, as this is being benchmarked, speed is of the essence. Instead of simply creating a single process per image, the solution uses a thread safe queue to break-up tasks for each image. As soon as a task is finished on an image it can pass to another task on the same or another image depending on what's next in the queue. 

On my laptop (i5-4300U with 4 cores), I get times of about 6sec for each axis on the 7 pictures provided.

You can choose on the command line what axis should be processed with the -t parameter (accepts : x, y or both). Files are outputed to 'processed_images' sub directory with naming nomenclature as specified.

Comments for specific points are within the code.

### CODE REVIEW

**Comment by poudro** [comment]
A little extra comment : I may have forgotten to check that the number of pictures being processed is indeed greater or equal to the number of CPU cores. 

The Pool should be initialized with min(n_cpus, n_pics), (there is no real point in this case to have more processes than the number of pictures)

File: app.py [hunk 1]
```diff
@@ -60,12 +65,131 @@ def crop_vmiddle(cls, name, image, height=300):
...
+
+"""set of tasks to perform in order on each image
+the 'reset' is simply to reset operations when applying on both x-scaled
+and y-scaled axii
+
+use one of the three following lists to perform desired operations, the parameter
+is set on command line """
+x_task_pool = [BusbudBanner.load, BusbudBanner.scale_x, BusbudBanner.blur, crop_and_save_x]
+y_task_pool = [BusbudBanner.load, BusbudBanner.scale_y, BusbudBanner.blur, crop_and_save_y]
+both_task_pool = [BusbudBanner.load, BusbudBanner.scale_x, BusbudBanner.blur, crop_and_save_x, 'reset', BusbudBanner.load, BusbudBanner.scale_y, BusbudBanner.blur, crop_and_save_y]
```
Comment (meunierd) [inline app.py:141]: http://legacy.python.org/dev/peps/pep-0008/#maximum-line-length

File: app.py [hunk 2]
```diff
@@ -60,12 +65,131 @@ def crop_vmiddle(cls, name, image, height=300):
...
+task_pool = []
+
+
+"""this function gets the current position in the queue, applies the desired
+function from task_pool and puts a new tuple at the end of queue for the
+next task on file.
+
+The idea is to be able to control the sequence of calls on each file without
+waiting for other files to finish each step, everything happens in parallel"""
```
Comment (meunierd) [inline app.py:150]: http://legacy.python.org/dev/peps/pep-0257/#multi-line-docstrings

File: app.py [hunk 3]
```diff
@@ -1,14 +1,19 @@
 # -*- coding: utf8 -*-
 
-import os
-
+import os, multiprocessing, StringIO
+import argparse, sys
```
Comment (meunierd) [inline app.py:4]: http://legacy.python.org/dev/peps/pep-0008/#imports

**Pull request closed without merging**